gnovm/pkg/gnolang/types.go

```diff
@@ -2098,6 +2108,44 @@ func assertSameTypes(lt, rt Type) {
 	}
 }
 
+func isEqualityTypes(lt, rt Type) bool {
+	debugPP.Printf("check isEqualityTypes, lt: %v, rt: %v, isLeftDataByte: %v, isRightDataByte: %v \n", lt, rt, isDataByte(lt), isDataByte(rt))
+	if lpt, ok := lt.(*PointerType); ok { // refer to 0f31
+		lt = lpt.Elt
+	}
+	if rpt, ok := rt.(*PointerType); ok {
+		rt = rpt.Elt
+	}
+	// lt or rt could be nil in runtime, e.g. a == nil, type of RHS would be nil
+	if lt == nil && rt == nil {
+		// both are nil.
+	} else if lt == nil || rt == nil {
+		// one is nil.  see function comment.
+	} else if lt.Kind() == rt.Kind() &&
+		isUntyped(lt) || isUntyped(rt) { // XXX, is this necessary?
+		// one is untyped of same kind.
+	} else if lt.Kind() == rt.Kind() &&
+		isDataByte(lt) { // XXX, what hit this
+		// left is databyte of same kind,
+		// specifically for assignments.
+		// TODO: make another function
+		// and remove this case?
+	} else if lt.Kind() == InterfaceKind &&
+		IsImplementedBy(lt, rt) {
```

Comment:
This definitely doesn't match the function name... not about equality.
But also does this ever happen? LT.Kind() should always be concrete for == binary operators, not an interface.

Reply:
this is erroneously introduced, fixed.